spec.md

```diff
@@ -901,10 +905,25 @@ reg myreg: SInt, myclock with: (reset => (myreset, myinit))
 ; ...
 ```
 
-Note that the clock signal for a register must be of type `clock`{.firrtl},
-the reset signal must be a `Reset`{.firrtl}, `UInt<1>`{.firrtl}, or
-`AsyncReset`{.firrtl}, and the type of initialization value must be equivalent
-to the declared type of the register (see [@sec:type-equivalence] for details).
+Optionally, a register can be declared with an enable signal. In the following
+example, `myreg`{.firrtl} is written the value `myval`{.firrtl} when the signal
+`myen`{.firrtl} is high.  Enable must be a value of type UInt<1>.
```

Comment:
should we define what the enable signal means (value is kept unless enable is true) or is that outside the scope of this spec (which doesn't explain what a register is, it seems to assume the reader already knows this)?

Reply:
we should, I had a couple things lost in a rebase, let me revise.

Reply:
Also the relationship between reset and enable.

Reply:
Agree. Define the behavior, i.e., "hold previous value".

Reply:
Should enable be a `UInt<1>` or an unknown width that infers to a `UInt<1>`? I prefer the latter interpretation.